external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts

```diff
@@ -187,6 +188,27 @@ export class MoveDebugSession extends LoggingDebugSession {
         this.sendEvent(new InitializedEvent());
     }
 
+    /**
+     * Intercepts all requests sent to the debug adapter to handle custom ones.
+     *
+     * @param request request to be dispatched.
+     */
+    protected dispatchRequest(request: DebugProtocol.Request): void {
```

Comment:
Have I mentioned how much I love OO programming? Reading this and seeing `super` is...yea lol.

<img src="https://media4.giphy.com/media/3oKHWn2hsiAKF0xIsM/giphy.gif"/>